### ISSUE TITLE
EnableBlinking and DisableBlinking don't work in the GNOME terminal

### ISSUE THREAD

**Issue by Cl00e9ment** [crossterm-rs/crossterm#650]
**Describe the bug**
`EnableBlinking` and `DisableBlinking` don't work in the GNOME terminal.

**Proposition**
The control sequences for enabling/disabling blinking (`\x1b[?12h` and `\x1b[?12l`) don't seem to work in some terminals, however the control sequences that set the shape and the blinking state at the same time (`\x1b[x q`, with `x` an integer) are working.

Also, `SetCursorShape` modify the blinking state with an hard-coded value (block is steady, and underline and line are blinking).

Since the `EnableBlinking` and `DisableBlinking` commands are not working in some terminals and that `SetCursorShape` modify the blinking state with an hard-coded value, why not choosing the cursor shape and blinking state at the same time and making: `EnableBlinking`, `DisableBlinking` and `SetCursorShape` deprecated?

We can imagine a `SetCursorStyle` command that look like this:

```rust
pub struct SetCursorStyle(pub CursorStyle);

impl Command for SetCursorStyle {
    fn write_ansi(&self, f: &mut impl fmt::Write) -> fmt::Result {
        use CursorStyle::*;
        match self.0 {
            BlinkingBlock => f.write_str("\x1b[?12h\x1b[0 q"),
            SteadyBlock => f.write_str("\x1b[?12l\x1b[2 q"),
            BlinkingUnderScore => f.write_str("\x1b[?12h\x1b[3 q"),
            SteadyUnderScore => f.write_str("\x1b[?12l\x1b[4 q"),
            BlinkingBar => f.write_str("\x1b[?12h\x1b[5 q"),
            SteadyBar => f.write_str("\x1b[?12l\x1b[6 q"),
        }
    }

    #[cfg(windows)]
    fn execute_winapi(&self) -> Result<()> {
        Ok(())
    }
}
```

Note that I'm using the `\x1b[?12h` and `\x1b[?12l` control sequences as a fail-over, in case `\x1b[x q` isn't supported.
If you think that it's a good idea, I can make the PR.

**OS**
 - Manjaro Linux

**Terminal/Console**
 - GNOME Terminal 3.42.2 for GNOME 41

**Comment by peepopoggers** [comment]
Can confirm.

**Comment by notfirefox** [comment]
I think that such a pull request would be very welcome. I'm having a similar problem using Konsole on KDE.

**Comment by TimonPost** [comment]
Yes this would be a nice to have. Little redundancy in `ESC[?12h` for blinking and then using `ESC [ _n_ SP q` for an other type of blinking could perhaps assure that blinking happens.

**Comment by TimonPost** [comment]
Fixed by: #742
